homeassistant/components/recorder/core.py

```diff
@@ -965,26 +983,41 @@ def _migrate_schema_and_setup_run(
             "recorder_database_migration",
         )
         self.hass.add_job(self._async_migration_started)
+        try:
+            migration_result, schema_status = self._migrate_schema(schema_status, True)
+            if migration_result:
+                self._setup_run()
+            return migration_result, schema_status
+        finally:
+            self.migration_in_progress = False
+            persistent_notification.dismiss(self.hass, "recorder_database_migration")
 
+    def _migrate_schema(
```

Comment:
The variable naming confused me at first when I read this since is returns a new schema_status and takes schema_status.
I might write this like

```python
    def _migrate_schema(
        self,
        schema_status: migration.SchemaValidationStatus,
        live: bool,
    ) -> tuple[bool, migration.SchemaValidationStatus]:
        """Migrate schema to the latest version."""
        assert self.engine is not None
        try:
            if live:
                migrator = migration.migrate_schema_live
            else:
                migrator = migration.migrate_schema_non_live
            new_schema_status = migrator(
                self, self.hass, self.engine, self.get_session, schema_status
            )
        except exc.DatabaseError as err:
            if self._handle_database_error(err):
                return (True, new_schema_status)
            _LOGGER.exception("Database error during schema migration")
            return (False, new_schema_status)
        except Exception:
            _LOGGER.exception("Error during schema migration")
            return (False, new_schema_status)

        return (True, new_schema_status)
```

Reply:
Good suggestion, fixed